magento/pwa-studio
# Requires and trims gift card input

## Description

This PR adds validation to and trims the gift card input field in the "Gift Cards" section of the `/cart` page.

## Related Issue





Closes PWA-374.

## Acceptance 



### Verification Stakeholders

### Specification



### Verification Steps



1. Add an item to your cart
1. Navigate to the `/cart` page

#### Validation

1. Without typing anything into the gift card input field, click "Check Balance"
1. Verify a validation message appears
1. Populate the field and remove focus from it
1. Verify the validation message disappears
1. Clear out the field and click "Apply"
1. Verify that the validation message appears

#### Trimming

1. Input "    02Y0LCHUNDRZ" (note the prepended whitespace) and click "Check Balance"
1. Verify the request is successful and the UI updates with the balance
1. Click "Apply"
1. Verify the request is successful and the card is applied

## Screenshots / Screen Captures (if appropriate)

![Screen Shot 2020-02-19 at 9 14 31 AM](https://user-images.githubusercontent.com/13182778/74852592-68b5b400-5302-11ea-815c-b5b3845e54f0.png)
**The new validation message**

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.
